homeassistant/components/octoprint/config_flow.py

```diff
@@ -269,7 +268,15 @@ def _get_octoprint_client(self, user_input: dict) -> OctoprintClient:
             ssl=False if not verify_ssl else None,
         )
         session = aiohttp.ClientSession(connector=connector)
-        _async_register_default_clientsession_shutdown(self.hass, session)
+
+        @callback
+        def _async_close_websession(event: Event) -> None:
+            """Close websession."""
+            session.detach()
+
+        self.hass.bus.async_listen_once(
```

Comment:
Implement the `async_remove` method in the flow instead and close the session in that method. It will run when the flow is aborted or on create entry result.

Reply:
There are multiple places where a session is created, so I added a list to remember all of them. Reworking the flow to only use one would probably be possible, but this feels way out of scope for this issue.